src/plugins/intel_npu/src/plugin/src/compiled_model.cpp

```diff
@@ -327,4 +344,23 @@ void CompiledModel::initialize_properties() {
     }
 }
 
+void CompiledModel::create_executor() {
+    const bool configCreateExecutor = _config.get<CREATE_EXECUTOR>();
+    static const auto envVar = std::getenv("IE_NPU_CREATE_EXECUTOR");
+    const bool IE_NPU_CREATE_EXECUTOR =
+        envVar ? envVarStrToBool("IE_NPU_CREATE_EXECUTOR", envVar) : configCreateExecutor;
```

Comment:
Since you touch this method, we can consider a small refactoring.

`CREATE_EXECUTOR` config overloads `envVar` method, so handling of `IE_NPU_CREATE_EXECUTOR` can be done by `Config` implementation. Can you consider avoid parsing `IE_NPU_CREATE_EXECUTOR` and rely on parsing from `Config`?

If this env variable is required in Release configuration, we can remove ifdef macro.

Reply:
Good one, thank you, I've brought the change in the last commit. We've chosen to remove the environment variable from the Release configuration, at least for now, since:
* Nowadays almost all jobs rely on the CiD compilation. The only real use of the variable in the Release configuration was to avoid the interaction with the driver when only compilation was desired in the CiP scenario. The CiD case implies the driver interaction anyway.
* Allowing the use of environment variables inside Release executables is something we might not want to allow. This information is not yet confirmed.